### PR TITLE
CFD-284 Homepage : Create layout (panel) 

### PR DESCRIPTION
**Pull request by HelenaEksler**
#455

### CODE REVIEW

**Comment by HelenaEksler** [comment]
@nader77 You can checkout from this branch.

**Comment by HelenaEksler** [comment]
selection_198

Homepage for Anonymous user:

selection_199

**Comment by HelenaEksler** [comment]
@amitaibu Could you please review this? 

1. Created new page - homepage.
2. Created custom layout for homepage for anonymous user.
3. Created content type `page_element`
4. Created view `Intro block` for page elements for page - `homepage` and with location `top`.
5. Show Intro block on the homepage on the variant for Anonymous users.

**Comment by HelenaEksler** [comment]
Update: 
1. Homage - is a variant of the group dashboard page.
2. Need to be added migration of the page elements (for Intro block)
3. Now when user doesn't have access to the group dashboard - instead of "Page not found" he sees the homepage.

**Comment by zero2one** [comment]
@HelenaEksler 

Just to be clear:

Homepage is not a variant of the group dashboard.
BUT
It sits on the same path (/)

So:
* Create a new Custom Page "Homepage" (page-c4m_homepage)
  * Path : **<front>**
  * No active group context
* Create 3 variants for it:
  * Anonymous user.
  * Logged in user **without** group memberships.
  * Logged in user **with** group memberships.
* Put the config in a new module called:
  * **c4m_features_homepage**

**Comment by HelenaEksler** [comment]
Update: 
1. Homage - is a variant of the group dashboard page.
2. For anonymous user:
- Homepage - variant of the homepage for the anonymous user
- Public group dashboard - can see the dashboard
- Private/Restricted group dashboard - can see Access denied.

3. For administrator:
- Homepage - variant of the homepage for the authenticated user
- Public group dashboard - can see the dashboard
- Private/Restricted group dashboard - can see Access denied.

4. For administrator:
- Homepage - variant of the homepage for the authenticated user
- Can see all group dashboards

5. Added migration of the page elements (for Intro block)

@amitaibu Could you review this, please?

**Comment by HelenaEksler** [comment]
@nader77 Review this, please.

File: capacity4more/modules/c4m/content/c4m_content_page_element/c4m_content_page_element.features.field_base.inc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ */
+
+/**
+ * Implements hook_field_default_field_bases().
+ */
+function c4m_content_page_element_field_default_field_bases() {
+  $field_bases = array();
+
+  // Exported field_base: 'body'
+  $field_bases['body'] = array(
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_page_element/c4m_content_page_element.features.field_base.inc]: ``body`` => ``c4m_body``, to be consistent with all the other fields.

**Comment by nader77** [comment]
@HelenaEksler Could you resize the ``home_anonymous.png``, it's huge, doesn't need to be more than 100px in width.

**Comment by nader77** [comment]
@HelenaEksler This is ready right ?

**Comment by HelenaEksler** [comment]
@nader77 Yes.

File: capacity4more/modules/c4m/content/c4m_content_page_element/c4m_content_page_element.features.field_base.inc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+/**
+ * Implements hook_field_default_field_bases().
+ */
+function c4m_content_page_element_field_default_field_bases() {
+  $field_bases = array();
+
+  // Exported field_base: 'field_page'
+  $field_bases['field_page'] = array(
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_page_element/c4m_content_page_element.features.field_base.inc]: ``field_page`` => ``c4m_page``, And for all the other fields.

**Comment by nader77** [comment]
@HelenaEksler Installed and review, When fields are fixed and green, You can assign to Peter.

**Comment by zero2one** [comment]
Tests are green, assigned myself.

**Comment by HelenaEksler** [comment]
@zero2one I just haven't changed the fields names. I can do this tomorrow in the morning or in the following issue.

**Comment by zero2one** [comment]
@HelenaEksler assigned back to you…

File: capacity4more/themes/c4m/kapablo/layouts/c4m/homepage/homepage.tpl.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * @ingroup themeable
+ */
+?>
+
+<div class="main-container container">
+
+  <div class="row">
+    <div class="col-md-12">
+      <?php print $content['content_top']; ?>
+      <input type="button" onclick="window.location.replace('user/register')" value="JOIN - SHARE - LEARN">
```
Comment (zero2one) [inline capacity4more/themes/c4m/kapablo/layouts/c4m/homepage/homepage.tpl.php]: The button should not be in the layout, it should be in the content of the "Join. Share. Learn." block.

**Comment by zero2one** [comment]
@HelenaEksler 

The global homepage variants are now part of the "group dashboard" module.
That's not correct (as we can't have a site without homepage, but we can have a site without groups).

> We need to put the frontpage page in a new module called c4m_features_homepage.

So do this:
1. Clone the existing page **page-c4m_og_group_dashboard** into **page-c4m_homepage**
2. Delete the existing page **page-c4m_og_group_dashboard**
3. Export the page **page-c4m_homepage** into a new module called **c4m_features_homepage**.
4. Export the page variant **Homepage Anonymous** into the new module (c4m_features_homepage).
5. Export the page variant **Homepage Authenticated** into the new module (c4m_features_homepage).
6. Export the page variant **Group Dashboard** into existing module **c4m_features_og_group_dashboard**
7. Export the page variant **access denied** into existing module **c4m_features_og_group_dashboard**

**Comment by zero2one** [comment]
Pull in this branch also the Pull request that have a dependency on this PR:

* #458 - CFD-290 : Homepage : Activity stream (Anonymous users)
* #460 - CFD-310 : Homepage : Create new content type "Featured" 
* #463 - CFD-289 : Homepage : Upcoming Events (Anonymous users)

**Comment by nader77** [comment]
This PR has a dependency as well:

* #476 - CFD-311 : Homepage: Add video to header about "what is C4D"

**Comment by HelenaEksler** [comment]
@zero2one Now the whole front page (both homepage and group dashboard) is in the homepage module. I haven't find the way to separate the variants for the custom page. So If it's a must - lets try to do this in the following PR in order not to stop the whole process. 

@nader Please, re-roll and merge checked PRs that depend on this brunch into it.

**Comment by nader77** [comment]
@HelenaEksler I merged the activity stream branch to this, The only branch left is #476 which can be ignored if you wanna review and merge this to the main branch.

**Comment by HelenaEksler** [comment]
@zero2one You can merge this. If there are still some changes you want to be done - create for it following issues, please.

**Pull request merged**